autorest/preparer.go

```diff
@@ -197,6 +203,66 @@ func WithFormData(v url.Values) PrepareDecorator {
 	}
 }
 
+// WithMultiPartFormData returns a PrepareDecoratore that "URL encodes" (e.g., bar=baz&foo=quux) form parameters
+// into the http.Request body.
+func WithMultiPartFormData(formDataParameters map[string]interface{}) PrepareDecorator {
+	return func(p Preparer) Preparer {
+		return PreparerFunc(func(r *http.Request) (*http.Request, error) {
+			r, err := p.Prepare(r)
+			if err == nil {
+				body := &bytes.Buffer{}
+				writer := multipart.NewWriter(body)
+				for key, value := range formDataParameters {
+					switch v := value.(type) {
```

Comment:
no need to `switch := ...(type)` here. just use `if rc, ok := v.(io.ReadCloser); ok { ...} else { ...}`